web/api/v1/api.go

```diff
@@ -147,6 +148,8 @@ func (api *API) Register(r *route.Router) {
 
 	r.Get("/targets", instr("targets", api.targets))
 	r.Get("/alertmanagers", instr("alertmanagers", api.alertmanagers))
+
+	r.Get("/read", prometheus.InstrumentHandler("read", http.HandlerFunc(api.remoteRead)))
```

Comment:
If we can agree for it to be temporary and to be removed at any time within 2.x, I guess it's alright.